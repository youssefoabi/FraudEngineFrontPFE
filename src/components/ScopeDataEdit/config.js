// eslint-disable-next-line import/prefer-default-export
export const config = [
  {
    columns: 4,
    id: 'name',
    type: 'input',
    label: 'Nom de la gamme',
  },
  {
    columns: 12,
    id: 'urlTemplate',
    type: 'input',
    label: 'Url avec param√®tres',
  },
];
