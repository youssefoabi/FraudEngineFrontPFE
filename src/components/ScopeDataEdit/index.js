import { DataEdit, GridItem } from '@cdiscount/backoffice-ui';
import { func, number, shape, string } from 'prop-types';
import React from 'react';

import { config } from './config';

const ScopeDataEdit = ({ scope, handleChange }) => (
  <GridItem title="CaractÃ©ristiques de la gamme">
    <DataEdit fields={config} values={scope} onChange={handleChange} />
  </GridItem>
);

export const scopeShape = shape({
  id: number,
  modifiedAt: number,
  name: string,
  urlTemplate: string,
});

ScopeDataEdit.propTypes = {
  handleChange: func,
  scope: scopeShape,
};

ScopeDataEdit.defaultProps = {
  handleChange: Function.prototype,
  scope: null,
};

export default ScopeDataEdit;
