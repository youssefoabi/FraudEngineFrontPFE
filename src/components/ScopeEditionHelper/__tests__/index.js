import { itRendersAllMutations } from '@cdiscount/jest-utils';

import ScopeDataEdit from '../';

const mutations = [
  {
    name: 'without props',
    props: {},
  },
  {
    name: 'with scope',
    props: {
      scope: {
        name: 'scope name',
        // eslint-disable-next-line @cdiscount/cdiscount/relative-protocol
        urlTemplate: 'https://www.{cdiscount_env}.com/{path_var}?{query_param=}',
      },
    },
  },
];
describe(ScopeDataEdit.name, () => {
  itRendersAllMutations(ScopeDataEdit, mutations);
});
