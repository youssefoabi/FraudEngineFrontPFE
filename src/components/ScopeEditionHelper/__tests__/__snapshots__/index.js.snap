// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScopeEditionHelper ScopeEditionHelper renders correctly with scope 1`] = `
<h
  fullheight={false}
  title="Aperçu de l'url"
  xs={12}
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-linkTypography-3"
  >
    <WithStyles(ForwardRef(Link))
      className="makeStyles-linkAnchor-4"
      href="https://www..com/?query_param="
      onClick={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      https://www..com/?query_param=
    </WithStyles(ForwardRef(Link))>
  </WithStyles(ForwardRef(Typography))>
  <f
    config={null}
    fields={
      Array [
        Object {
          "columns": 7,
          "defaultValue": "",
          "id": "{cdiscount_env}",
          "label": "cdiscount_env",
          "type": "input",
        },
        Object {
          "columns": 7,
          "defaultValue": "",
          "id": "{path_var}",
          "label": "path_var",
          "type": "input",
        },
        Object {
          "columns": 7,
          "defaultValue": "",
          "id": "{query_param=}",
          "label": "query_param",
          "type": "input",
        },
      ]
    }
    inputs={null}
    onChange={[Function]}
    onInit={[Function]}
    title={null}
    values={
      Object {
        "{cdiscount_env}": "",
        "{path_var}": "",
        "{query_param=}": "",
      }
    }
  />
</h>
`;

exports[`ScopeEditionHelper ScopeEditionHelper renders correctly without props 1`] = `
<h
  fullheight={false}
  title="Aperçu de l'url"
  xs={12}
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-linkTypography-1"
  >
    <WithStyles(ForwardRef(Link))
      className="makeStyles-linkAnchor-2"
      href=""
      onClick={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    />
  </WithStyles(ForwardRef(Typography))>
  <f
    config={null}
    fields={Array []}
    inputs={null}
    onChange={[Function]}
    onInit={[Function]}
    title={null}
    values={Object {}}
  />
</h>
`;
